File: FileParserServer/relation.py
class Relation:
    r_id = 0

    def __init__(self, _p_id=0, _name="", _first=0, _second=0):
        self.id = Relation.r_id
        Relation.r_id += 1

        self.name = _name
        self.the_first_e_id = _first
        self.the_second_e_id = _second
        self.p_id = _p_id
        self.source_id = _first.id
        self.target_id = _second.id
        self.source = _first.name
        self.target = _second.name
        self.type = ""
        self.hash_id = self.id

    def __str__(self):
        return "id: " + str(self.id) + ", name: " + self.name + ", first: " + str(self.the_first_e_id) + ", second: " + str(self.the_second_e_id)

    def to_dic(self):
        dic = {"rid": self.id, "pid": self.p_id, "source_id": self.source_id,
               "target_id": self.target_id, "source": self.source, "target": self.target,
               "relation": self.name, "type": self.type, "hash_id": self.hash_id}
        return dic

File: SpringProject/src/main/java/com/example/demo/po/Relation.java
package com.example.demo.po;

import lombok.Data;

/**
 * @Author: Owen
 * @Date: 2021/3/7 14:40
 * @Description:
 */
@Data
public class Relation {

    private Integer rid;

    private Integer pid;

    private String source_id;

    private String target_id;

    private String source;

    private String target;

    private String relation;

    private String type;

    private String hash_id;

    public Relation(String source_id, String target_id, String source, String target,
                    String relation, String hash_id){
        this.source_id = source_id;
        this.target_id = target_id;
        this.source = source;
        this.target = target;
        this.relation = relation;
        this.hash_id = hash_id;
    }

    public Relation(Integer pid, String source_id, String target_id, String source, String target,
                    String relation, String hash_id){
        this.pid = pid;
        this.source_id = source_id;
        this.target_id = target_id;
        this.source = source;
        this.target = target;
        this.relation = relation;
        this.hash_id = hash_id;
    }

    public Relation(){
    }

    public Integer getRid() {
        return rid;
    }

    public void setRid(Integer rid) {
        this.rid = rid;
    }

    public Integer getPid() {
        return pid;
    }

    public void setPid(Integer pid) {
        this.pid = pid;
    }

    public String getSource_id() {
        return source_id;
    }

    public void setSource_id(String source_id) {
        this.source_id = source_id;
    }

    public String getTarget_id() {
        return target_id;
    }

    public void setTarget_id(String target_id) {
        this.target_id = target_id;
    }

    public String getSource() {
        return source;
    }

    public void setSource(String source) {
        this.source = source;
    }

    public String getTarget() {
        return target;
    }

    public void setTarget(String target) {
        this.target = target;
    }

    public String getRelation() {
        return relation;
    }

    public void setRelation(String relation) {
        this.relation = relation;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getHash_id() {
        return hash_id;
    }

    public void setHash_id(String hash_id) {
        this.hash_id = hash_id;
    }
}
